app/src/org/commcare/views/notifications/NotificationMessage.java

```diff
@@ -44,15 +53,23 @@ public NotificationMessage[] newArray(int size) {
         }
     };
 
-    public NotificationMessage(String context, String title, String details, String action, Date date) {
+    public NotificationMessage(String context, String title, String details, String action, Date date)  {
+        this(context, title, details, action, date, null);
+    }
+
+    public NotificationMessage(String context, String title, String details, String action, Date date, NotificationActionButtonInfo buttonInfo) {
         if (context == null || title == null || details == null || date == null) {
-            throw new NullPointerException("None of the arguments for creating a NotificationMessage may be null except for action");
+            throw new NullPointerException("None of the arguments for creating a NotificationMessage may be null except for action and buttonInfo");
+        }
+        if (buttonInfo != null && buttonInfo.getButtonText() == null) {
+            throw new NullPointerException("NotificationMessage with buttonInfo can not have null button text");

```

Comment:
move it to the buttonInfo constructor itself ?

Reply:
Agreed